waves/src/components/AssetSelect.tsx

```diff
@@ -0,0 +1,175 @@
+import { ChevronDownIcon } from "@chakra-ui/icons";
+import {
+    Box,
+    Button,
+    Drawer,
+    DrawerBody,
+    DrawerContent,
+    DrawerHeader,
+    DrawerOverlay,
+    Grid,
+    HStack,
+    Image,
+    Modal,
+    ModalBody,
+    ModalCloseButton,
+    ModalContent,
+    ModalFooter,
+    ModalHeader,
+    ModalOverlay,
+    Text,
+    useDisclosure,
+    VStack,
+} from "@chakra-ui/react";
+import React, { MouseEvent } from "react";
+import { GrGithub } from "react-icons/gr";
+import { AssetType } from "../App";
+import Bitcoin from "./bitcoin.svg";
+import Xmr from "./monero.svg";
+import Usdt from "./tether.svg";
+
+interface CurrencySelectProps {
+    type: AssetType;
+    onAssetChange: (asset: AssetType) => void;
+    placement: "left" | "right";
+}
+
+function AssetSelect({ type, onAssetChange, placement }: CurrencySelectProps) {
+    const onChange = (value: AssetType) => {
+        onAssetChange(value);
+        onClose();
+    };
+    const btnRef = React.useRef(null);
+
+    const { isOpen, onOpen, onClose } = useDisclosure();
+
+    return (
+        <>
+            <Button ref={btnRef} w="100%" bg="white" border="grey" size="lg" shadow="md" onClick={onOpen}>
+                {type === "BTC" && <BitcoinSelect />}
+                {type === "USDT" && <UsdtSelect />}
+            </Button>
+
+            <Drawer placement={placement} onClose={onClose} isOpen={isOpen} size="sm">
+                <DrawerOverlay>
+                    <DrawerContent>
+                        <DrawerHeader>{"Available Assets"}</DrawerHeader>
+                        <DrawerBody>
+                            <Grid templateColumns="repeat(2, 1fr)" gap={6}>
+                                <BitcoinBox onSelect={onChange} />
+                                <UsdtBox onSelect={onChange} />
+                                <XmrBox />
```

Comment:
On the contrary, I think it's a very good idea to get more attention to our other projects :) We might even go one step further and integrate @da-kami's mock which lead to the commandline interface.

Reply:
Thanks for sharing your opinion! Can you please also address the points I am bringing up of _why_ I think it is not a good idea?

To be clear, I think it is definitely a good idea to get more attention to our other projects. However, what bugs me are two things:

1. We are introducing additional complexity into the UX to achieve this.

To be more precise, I think you can get rid of the whole asset selector concept in this prototype because all we support are two assets (if we take the advertisement out). That will simplify the UX because there are less elements to interact with. All you need is a button in the middle to flip the direction.

2. It is misleading the user into thinking that this functionality will be available soonish / eventually.

Given all plans / ideas I was exposed to so far, this is not on our radar.

---

The appropriate place for such things would be links in the footer of the page. That will also help with SEO because these links are permanently visible to search engines compared to a user invoked popup.

Putting them right in front of the user while they are trying to achieve a task is just distracting.

Reply:
Sorry, I did not want to ignore your opinion. 

Regarding misleading: 
The frontend here is only for Liquid (at least for now). I disagree with your opinion that this is misleading. It is brutal clickbait but a valid approach to get the user's attention. If we get feedback that they hate this or find it distracting we can remove it at any point of time. 

Regarding the complexity: fair point, it was also not that straight forward to design this drawer. However, the initial mock had a drop down with a currency selection and this was not criticized back then. I went ahead and implemented that exact mock as much as I was able to (chakra dropdowns look terrible, hence I chose a drawer instead).

I liked the idea of flipping sides using the button in the middle and went ahead and implemented it :)

Reply:
> It is brutal clickbait but a valid approach to get the user's attention.

I agree. It is likely an extremely effective tool in getting the user's attention. That is exactly my point! When interacting with the asset dropdowns / drawers, the user is in the middle of the primary task that this tool exposes: buying / selling L-BTC.

The user completing this task is equivalent to a checkout page on an e-commerce site. If it is completed, the merchant will earn money. As such, anything that distracts the user from completing it is equivalent to lost money.

Amazon for example even disables the link behind its logo once you are in the checkout process to make it purposely harder for you to leave the page. I can only assume they do so because it increases revenue.

> However, the initial mock had a drop down with a currency selection and this was not criticized back then.

I didn't think about it in such detail back then to be honest. But the mock also didn't have assets in them that you couldn't actually swap.